src/sql/workbench/contrib/query/browser/queryEditor.ts

```diff
@@ -186,6 +186,12 @@ export class QueryEditor extends BaseEditor {
 		this._actualQueryPlanAction = this.instantiationService.createInstance(actions.ActualQueryPlanAction, this);
 		this._toggleSqlcmdMode = this.instantiationService.createInstance(actions.ToggleSqlCmdModeAction, this, false);
 		this._exportAsNotebookAction = this.instantiationService.createInstance(actions.ExportAsNotebookAction, this);
+		this.setTaskbarContent();
```

Comment:
setTaskbarContent() is being called twice now, is that intended?

Reply:
My understanding is yes - because she still needs to update the actions once a connection is set so the Kusto actions are properly set. But see my comment for another concern I have for the Kusto side of this

Reply:
Yes, in case of connection we decide the query actions based on the connection provider i.e. Kusto connections will have different query actions